File: FE/src/features/customer/services/customerService.ts
import { apiClient } from '../../../services/apiClient'
import type { PaginatedResult } from '../../../types/api.types'
import type {
  CreateCustomerRequest,
  CustomerResponse,
  UpdateCustomerRequest,
} from '../types/customer.types'

export const customerService = {
  getAll: (page = 1, size = 10, sort = 'id,asc', active?: boolean, q?: string) => {
    const params = new URLSearchParams({
      page: String(page),
      size: String(size),
      sort,
    })
    if (active !== undefined) params.set('active', String(active))
    if (q?.trim()) params.set('q', q.trim())
    return apiClient<PaginatedResult<CustomerResponse>>(`/customers?${params}`)
  },

  getById: (id: number) => apiClient<CustomerResponse>(`/customers/${id}`),

  create: (data: CreateCustomerRequest) =>
    apiClient<CustomerResponse>('/customers', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  update: (data: UpdateCustomerRequest) =>
    apiClient<CustomerResponse>('/customers', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  disable: (id: number) =>
    apiClient<CustomerResponse>(`/customers/${id}`, { method: 'DELETE' }),

  enable: (id: number) =>
    apiClient<CustomerResponse>(`/customers/${id}/enable`, { method: 'POST' }),
}

File: BE/src/main/java/Manager_vnd/Manager/feature/customer/CustomerService.java
package Manager_vnd.Manager.feature.customer;

import Manager_vnd.Manager.dto.PaginatedResult;
import Manager_vnd.Manager.feature.customer.dto.CreateCustomerRequest;
import Manager_vnd.Manager.feature.customer.dto.CustomerResponse;
import Manager_vnd.Manager.feature.customer.dto.UpdateCustomerRequest;

public interface CustomerService {

    PaginatedResult<CustomerResponse> getAll(int page, int size, String sort, Boolean active, String q);

    CustomerResponse getById(long id);

    CustomerResponse create(CreateCustomerRequest request);

    CustomerResponse update(UpdateCustomerRequest request);

    CustomerResponse disable(long id);

    CustomerResponse enable(long id);
}
